packages/core/src/components/tag/tag.tsx

```diff
@@ -31,11 +31,12 @@ export class Tag extends React.Component<ITagProps, {}> {
         const tagClasses = classNames(Classes.TAG, Classes.intentClass(intent), {
             [Classes.TAG_REMOVABLE]: onRemove != null,
         }, className);
+        const button = <button type="button" className={Classes.TAG_REMOVE} onClick={onRemove} />;
 
         return (
             <span {...removeNonHTMLProps(this.props)} className={tagClasses}>
                 {this.props.children}
-                {isFunction(onRemove) ? <button className={Classes.TAG_REMOVE} onClick={onRemove} /> : null}
+                {isFunction(onRemove) ? button : null}
```

Comment:
i meant pull the whole statement out. no need to create the object when the function is false. this should say just `{button}`

also please use `undefined` instead of `null`.